compute/client_library/ingredients/instances/ip_address/assign_static_ip_to_existing_vm.py

```diff
@@ -0,0 +1,93 @@
+#  Copyright 2024 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+# flake8: noqa
+
+
+import uuid
+
+from google.cloud.compute_v1 import InstancesClient
+from google.cloud.compute_v1.types import (
+    AccessConfig,
+    AddAccessConfigInstanceRequest,
+    DeleteAccessConfigInstanceRequest,
+)
+
+
+# <INGREDIENT assign_static_ip_to_existing_vm>
+def assign_static_ip_to_existing_vm(
+    project_id: str, zone: str, instance_name: str, ip_address: str
+):
+    """
+    Updates or creates an access configuration for a VM instance to assign a static external IP.
+    As network interface is immutable - deletion stage is required in case of any assigned ip (static or ephemeral).
+    VM and ip address must be created before calling this function.
+    IMPORTANT: VM and assigned IP must be in the same region.
```

Comment:
What if it's a global address?

Reply:
It is not supported for instances. Global IP addresses are designed for use with services that distribute traffic globally, such as global load balancers. Only regionals for instances